test/integration/targets/vmware_guest_disk/tasks/main.yml

```diff
@@ -77,3 +77,156 @@
   assert:
     that:
         - test_create_disk2 is changed
+
+- name: create new disk with custom shares
+  vmware_guest_disk:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: "{{ virtual_machines[0].name }}"
+    disk: 
+        - size_gb: 15
+          type: eagerzeroedthick
+          datastore: "{{ rw_datastore }}"
+          disk_mode: "independent_nonpersistent"
+          scsi_controller: 1
+          state: present
+          unit_number: 4
+          shares:
+            level: custom
+            level_value: 1300
+  register: test_custom_shares
+
+- debug:
+    msg: "{{ test_custom_shares }}"
+
+- name: assert that changes were made
+  assert:
+     that:
+         - test_custom_shares is changed
+ 
+- name: create new disk with custom IO limits and shares in IO Limits
+  vmware_guest_disk:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: "{{ virtual_machines[0].name }}"
+    disk: 
+        - size_gb: 16
```

Comment:
The same please.

Reply:
@goneri
as part of integration test we cant maintain 1gb of size in all test cases, as while updating the disk attributes, size should be more than the existing disk size , for now i am giving single digit disk sizes like 1,2,3,4